### PR TITLE
[wasm] Run browser tests on helix/windows

### PR DESCRIPTION
**Pull request by radekdoulik**
It updates the infrastructure to be able to run these.

It also uses updated docker image, which contains sans serif font for the chrome browser to be able to run. It was crashing without it.

### CODE REVIEW

**Comment by radekdoulik** [comment]
Oh, I still need to restore the matrix :-)

File: src/libraries/sendtohelixhelp.proj [hunk 1]
```diff
@@ -362,7 +375,7 @@
       <HelixWorkItem Include="@(_RunOnlyWorkItem -> '%(FileName)')" >
         <PayloadArchive>%(Identity)</PayloadArchive>
         <!-- No RunTests script generated for the sample project so we just use the direct command -->
-        <Command>$(ExecXHarnessCmd) wasm $(XHarnessCommand)  --app=. --browser=Chrome  --html-file=index.html --output-directory=$(XHarnessOutput) -- %(FileName).dll --testing</Command>
+        <Command>$(ExecXHarnessCmd) wasm $(XHarnessCommand)  --app=. --browser=Chrome $(XHarnessBrowserPath) --html-file=index.html --output-directory=$(XHarnessOutput) -- %(FileName).dll --testing</Command>
```
Comment (pavelsavara) [inline src/libraries/sendtohelixhelp.proj:378]: It seems that linux would find it on PATH, why windows are different ?

Comment (radekdoulik) [reply src/libraries/sendtohelixhelp.proj:378]: Probably some difference in xharness. At least it has the option this time. For JS engines I had to modify xharness to look for them.

File: src/tests/BuildWasmApps/Wasm.Build.Tests/Wasm.Build.Tests.csproj [hunk 2]
```diff
@@ -21,6 +21,25 @@
...
+
+  <PropertyGroup>
+    <_PreCommand Condition="'$(OS)' != 'Windows_NT'">$(_PreCommand) TEST_LOG_PATH=%24{XHARNESS_OUT}/logs</_PreCommand>
+    <_PreCommand Condition="'$(OS)' != 'Windows_NT'">$(_PreCommand) HARNESS_RUNNER=%24{HARNESS_RUNNER}</_PreCommand>
+    <_PreCommand Condition="'$(OS)' == 'Windows_NT'">$(_PreCommand) set TEST_LOG_PATH=%XHARNESS_OUT%\logs &amp;</_PreCommand>
+    <_PreCommand Condition="'$(OS)' == 'Windows_NT'">$(_PreCommand) set HARNESS_RUNNER=%HARNESS_RUNNER% &amp;</_PreCommand>
+
+    <RunScriptCommand Condition="'$(OS)' != 'Windows_NT'">$(_PreCommand) dotnet exec xunit.console.dll $(AssemblyName).dll -xml %24XHARNESS_OUT/testResults.xml</RunScriptCommand>
+    <RunScriptCommand Condition="'$(OS)' == 'Windows_NT'">$(_PreCommand) dotnet.exe exec xunit.console.dll $(AssemblyName).dll -xml %XHARNESS_OUT%\testResults.xml</RunScriptCommand>
+    <RunScriptCommand Condition="'$(ContinuousIntegrationBuild)' == 'true'">$(RunScriptCommand) -nocolor</RunScriptCommand>
```
Comment (pavelsavara) [inline src/tests/BuildWasmApps/Wasm.Build.Tests/Wasm.Build.Tests.csproj:39]: Duplicate line below ?

File: src/libraries/tests.proj [hunk 3]
```diff
@@ -318,6 +318,19 @@
...
+
+  <ItemGroup Condition="'$(TargetOS)' == 'Browser' and '$(BrowserHost)' == 'windows' and '$(Scenario)' == 'WasmTestOnBrowser' and '$(RunDisabledWasmTestsOnWindows)' != 'true'">
+    <!-- Issue: https://github.com/dotnet/runtime/issues/55429 -->
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)Microsoft.VisualBasic.Core\tests\Microsoft.VisualBasic.Core.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.IO\tests\System.IO.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Linq.Expressions\tests\System.Linq.Expressions.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Memory\tests\System.Memory.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Runtime\tests\System.Runtime.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Private.Xml\tests\Writers\XmlWriterApi\System.Xml.RW.XmlWriterApi.Tests.csproj" />
+    <ProjectExclusions Include="$(MonoProjectRoot)sample\wasm\browser\Wasm.Browser.Sample.csproj" />
```
Comment (pavelsavara) [inline src/libraries/tests.proj:332]: Already in ItemGroup above

**Review by pavelsavara — APPROVED** [review]

File: src/tests/BuildWasmApps/Wasm.Build.Tests/Wasm.Build.Tests.csproj (hunk 2 above)
Comment (radekdoulik) [reply src/tests/BuildWasmApps/Wasm.Build.Tests/Wasm.Build.Tests.csproj:39]: Interesting, looks like it got here in one of the merges.

Comment (radekdoulik) [reply src/tests/BuildWasmApps/Wasm.Build.Tests/Wasm.Build.Tests.csproj]: Ah, these were to make the wasm build tests to run on windows. I removed them as https://github.com/dotnet/runtime/pull/54451 will handle that too

File: src/libraries/sendtohelixhelp.proj [hunk 4]
```diff
@@ -393,6 +405,7 @@
       <ExecXHarnessCmd>dotnet exec %XHARNESS_CLI_PATH%</ExecXHarnessCmd>
       <XHarnessOutput>%HELIX_WORKITEM_UPLOAD_ROOT%\xharness-output</XHarnessOutput>
       <XHarnessCommand>%XHARNESS_COMMAND%</XHarnessCommand>
+      <XHarnessBrowserPath>--browser-path=%HELIX_CORRELATION_PAYLOAD%\chrome-win\chrome.exe</XHarnessBrowserPath>
```
Comment (radical) [inline src/libraries/sendtohelixhelp.proj:408]: nit: Maybe name this `XHarnessBrowserPathArg`

**Review by radical — APPROVED** [review]

**Comment by lewing** [comment]
It is green!

**Review by radical — APPROVED** [review]

**Comment by radical** [comment]
The failing tests could be disabled.

**Review by steveisok — APPROVED** [review]

**Pull request merged**